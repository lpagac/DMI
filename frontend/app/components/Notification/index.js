/**
 *
 * Notification
 *
 */

import styled from 'styled-components';

const Notification = styled.h3`
  font-size: 30px;
  margin: 10px;
  background-color: ${props => (props.success ? '#70e000' : '#f94144')};
  opacity: 0.3;
  color: ${props => (props.success ? '#6c757d' : '#343a40')};
  padding: 10px 12px;
  border-radius: 5px;
`;

export default Notification;
