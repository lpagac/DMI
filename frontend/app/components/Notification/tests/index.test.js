/**
 *
 * Tests for Notification
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Notification from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<Notification />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Notification />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Notification />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
