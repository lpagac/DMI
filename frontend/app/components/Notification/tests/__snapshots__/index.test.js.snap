// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> Should render and match the snapshot 1`] = `
<h3
  class="Notification-sc-1wuue35-0 YfSYQ"
/>
`;
